core/types/bitmap.go

```diff
@@ -0,0 +1,94 @@
+package types
+
+import (
+	"fmt"
+	"io"
+	"math/big"
+	"math/bits"
+
+	"github.com/autonity/autonity/rlp"
+)
+
+type Bitmap big.Int
+
+func NewBitmap() *Bitmap {
+	return (*Bitmap)(new(big.Int))
+}
+
+func (b *Bitmap) Contains(c *Bitmap) bool {
+	cBig := (*big.Int)(c)
+	bBig := (*big.Int)(b)
+	return new(big.Int).And(bBig, cBig).Cmp(cBig) == 0
+}
+
+// Diff does the set minus operation
+// B \ C = B & ~C gives the elements that are in B but not in C
+func (b *Bitmap) Diff(c *Bitmap) *Bitmap {
+	cBig := (*big.Int)(c)
+	bBig := (*big.Int)(b)
+	diff := new(big.Int).And(bBig, new(big.Int).Not(cBig))
+	return (*Bitmap)(diff)
+}
```

Comment:
```suggestion
// Diff does the set minus operation
// B \ C = B ^ (B & C) gives the elements that are in B but not in C
func (b *Bitmap) Diff(c *Bitmap) *Bitmap {
	cBig := (*big.Int)(c)
	bBig := (*big.Int)(b)
	diff := new(big.Int).Xor(bBig, new(big.Int).And(bBig, cBig))
	return (*Bitmap)(diff)
}
```

To avoid negative big.Int